tests/components/dyson/test_climate.py

```diff
@@ -154,236 +136,259 @@ def _get_device_heat_on():
+def patch_platforms_fixture():
+    """Only set up the climate platform for the climate tests."""
+    with patch("homeassistant.components.dyson.DYSON_PLATFORMS", new=[DOMAIN]):
+        yield
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[_get_device_heat_on()],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_pure_hot_cool_link_set_mode(mocked_login, mocked_devices, hass):
+    """Test set climate mode."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    device = mocked_devices.return_value[0]
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_HVAC_MODE: HVAC_MODE_HEAT},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(heat_mode=HeatMode.HEAT_ON)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_HVAC_MODE: HVAC_MODE_COOL},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(heat_mode=HeatMode.HEAT_OFF)
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[_get_device_heat_on()],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_pure_hot_cool_link_set_fan(mocked_login, mocked_devices, hass):
+    """Test set climate fan."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    device = mocked_devices.return_value[0]
+    device.temp_unit = TEMP_CELSIUS
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_FAN_MODE: FAN_FOCUS},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(focus_mode=FocusMode.FOCUS_ON)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_FAN_MODE: FAN_DIFFUSE},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(focus_mode=FocusMode.FOCUS_OFF)
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[_get_device_heat_on()],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_pure_hot_cool_link_state(mocked_login, mocked_devices, hass):
+    """Test set climate temperature."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORT_FLAGS
+    assert state.attributes[ATTR_TEMPERATURE] == 23
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)